client/src/main/java/com/paypal/selion/internal/reports/model/BaseLog.java

```diff
@@ -13,47 +13,45 @@
 |  the specific language governing permissions and limitations under the License.                                     |
 \*-------------------------------------------------------------------------------------------------------------------*/
 
-package com.paypal.selion.reports.model;
+package com.paypal.selion.internal.reports.model;
 
 import org.testng.Reporter;
 
 import com.paypal.selion.logger.SeLionLogger;
-import com.paypal.selion.reports.runtime.WebReporter;
+import com.paypal.selion.reports.runtime.SeLionReporter;
 import com.paypal.test.utilities.logging.SimpleLogger;
 
 /**
  * This class serves as the base for all logging with respect to UI operations being done either on the browser or on a
  * native app in a simulator/device. Any functionality that intends to provide a reporting capability similar to
- * {@link WebReporter} should leverage this class for the basic functionalities and add up only customizations as and
+ * {@link SeLionReporter} should leverage this class for the basic functionalities and add up only customizations as and
  * where required.
  * 
+ * This is an internal class for use in Selion and clients of SeLion should have no need for referencing this public class.
+ * 
  */
-public abstract class AbstractLog {
+public class BaseLog {
 
     protected static SimpleLogger logger = SeLionLogger.getLogger();
-
-    protected String type;
+    /** custom message for the log */
     protected String msg;
+    /** file location of the screenshot associated with this log message */
     protected String screen;
+    /** the url of the current page.  This applies only to webtests and mobiletests that are running with safari, etc. */
     protected String location;
+    /** file location of the page source associated with this log message */
+    protected String href;
 
-    /**
-     * Add custom parsing of a String that represents a line in the log.
-     * 
-     * @param part
-     */
-    abstract protected void parse(String part);
+    public BaseLog() {
+    }
```

Comment:
The member variables are purposefully defaulting to null.  There is code checks for null which means that this log doesnt contain a screenshot or a pagesource.